assets/eip-4973/src/ERC4973.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.6;
+
+import {ERC165} from "openzeppelin-contracts/utils/introspection/ERC165.sol";
+import {Counters} from "openzeppelin-contracts/utils/Counters.sol";
+
+import {IERC721Metadata} from "./interfaces/IERC721Metadata.sol";
+import {IERC4973} from "./interfaces/IERC4973.sol";
+
+
+abstract contract ERC4973 is ERC165, IERC721Metadata, IERC4973 {
```

Comment:
Inheriting from ERC165 and IERC721Metadata might not be necessary as IERC4973 imposes it already.